src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -85,3 +86,30 @@ def java_sources(self):
         raise TargetDefinitionException(self, 'No such java target: {}'.format(spec))
       targets.append(target)
     return targets
+
+  def sources_snapshot(self, scheduler=None):
```

Comment:
Would it make sense to do this in https://github.com/pantsbuild/pants/blob/d72fb81e0a2b8be81d22846047aefd480016dfa9/src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py#L423-L426 instead?

While it's awkward to have the special case exist in the compiler, afaict there is exactly one codepath that cares about this (neither javac_compile nor rsc_compile could care), and it avoids the shenanigans with `Snapshot` merging.

Reply:
Yes! Will make that edit.